frontend/service/wikibase.service.js

```diff
@@ -41,20 +42,55 @@ export class WikibaseService {
     return QITEM_PATTERN
   }
 
-  getEntity (id, lang) {
+  async getEntity (id, lang) {
     const url = this.wbk.getEntities({
       ids: [id],
       language: [lang, 'en']
     })
-
     const entityHash = this.hashCode(`${id}-${lang}`)
-    const entry = this.getResultsFromCache(entityHash)
-    if (entry) {
-      return new Promise((resolve, reject) => {
-        return resolve(entry.value)
-      })
+    let entity
+
+    const CACHE_MAX_ENTRIES = 100
+    const CACHE_EXPIRATION_MILLIS = 120000
+    const currentItemCount = localStorage.length
+
+    // If the limit is reached (100 items), remove the oldest item
+    if (currentItemCount >= CACHE_MAX_ENTRIES) {
+      const oldestKey = localStorage.key(0)
```

Comment:
Let's discuss it, I think in this case we need a temporary persistence for the cache.
The _localStorage_ could be useful in other places.

Reply:
ok ill try implementing with vuex instead

Reply:
Just rollback these changes for now and only implement cache for https://github.com/PhiloBiblon/philobiblon-ui/pull/5#discussion_r1358454088
I propose you implement without any cache and after that we think about it again.